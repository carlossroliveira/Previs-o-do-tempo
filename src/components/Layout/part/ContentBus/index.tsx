// -------------------------------------------------
// Packages
// -------------------------------------------------
import React from 'react';
// -------------------------------------------------
// Components
// -------------------------------------------------
import { Button } from '../../../Button';
import { Input } from '../../../Input';
// -------------------------------------------------
// Styles
// -------------------------------------------------
import { ContainerSC } from './contentBusStyles';

export const ContentBus = (): JSX.Element => {
  return (
    <ContainerSC>
      <Input />
      <Button />
    </ContainerSC>
  );
};
