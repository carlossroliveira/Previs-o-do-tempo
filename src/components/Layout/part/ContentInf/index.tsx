// -------------------------------------------------
// Packages
// -------------------------------------------------
import React from 'react';
// -------------------------------------------------
// Components
// -------------------------------------------------
import { SwitchComponent } from '../../../SwitchComponent';
// -------------------------------------------------
// Styles
// -------------------------------------------------
import { ContainerSC, TitleSC } from './contentInfStyles';

export const ContentInf = (): JSX.Element => {
  return (
    <ContainerSC>
      <TitleSC>Tema</TitleSC>
      <SwitchComponent />
    </ContainerSC>
  );
};
