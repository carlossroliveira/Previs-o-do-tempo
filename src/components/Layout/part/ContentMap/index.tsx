// -------------------------------------------------
// Packages
// -------------------------------------------------
import React from 'react';
// -------------------------------------------------
// Components
// -------------------------------------------------

// -------------------------------------------------
// Styles
// -------------------------------------------------
import { ContainerSC } from './ContentMapStyles';
// -------------------------------------------------
// Types
// -------------------------------------------------

export const ContentMap = (): JSX.Element => {
  return <ContainerSC>MAPA</ContainerSC>;
};
