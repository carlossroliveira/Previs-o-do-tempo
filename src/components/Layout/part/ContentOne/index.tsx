// -------------------------------------------------
// Packages
// -------------------------------------------------
import React from 'react';
// -------------------------------------------------
// Components
// -------------------------------------------------

// -------------------------------------------------
// Styles
// -------------------------------------------------
import { ContainerSC } from './contentOneStyles';
// -------------------------------------------------
// Types
// -------------------------------------------------

export const ContentOne = (): JSX.Element => {
  return <ContainerSC>CONTENTONE</ContainerSC>;
};
