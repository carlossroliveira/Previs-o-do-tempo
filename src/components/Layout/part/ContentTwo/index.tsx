// -------------------------------------------------
// Packages
// -------------------------------------------------
import React from 'react';
// -------------------------------------------------
// Components
// -------------------------------------------------

// -------------------------------------------------
// Styles
// -------------------------------------------------
import { ContainerSC } from './contentTwoStyles';
// -------------------------------------------------
// Types
// -------------------------------------------------

export const ContentTwo = (): JSX.Element => {
  return <ContainerSC>CONTENTTWO</ContainerSC>;
};
