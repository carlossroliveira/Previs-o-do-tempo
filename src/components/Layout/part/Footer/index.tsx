// -------------------------------------------------
// Packages
// -------------------------------------------------
import React from 'react';
// -------------------------------------------------
// Components
// -------------------------------------------------

// -------------------------------------------------
// Styles
// -------------------------------------------------
import { ContainerSC } from './footerStyles';
// -------------------------------------------------
// Types
// -------------------------------------------------

export const Footer = (): JSX.Element => {
  return (
    <ContainerSC>
      Animações Animações Animações Animações Animações Animações Animações
      Animações Animações Animações Animações Animações Animações Animações
      Animações Animações Animações Animações Animações Animações Animações
      Animações Animações Animações Animações Animações Animações Animações
      Animações Animações Animações Animações Animações Animações Animações
      Animações Animações Animações Animações Animações Animações{' '}
    </ContainerSC>
  );
};
