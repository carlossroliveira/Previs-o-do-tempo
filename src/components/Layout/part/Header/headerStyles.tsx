import styled, { keyframes } from 'styled-components';

export const ContainerSC = styled.section`
  grid-area: header;
  margin: 0 0 0 4rem;

  @media (max-width: 500px) {
    margin: 0;
  }
`;

export const TextAnimation = keyframes`
  to {
      opacity: 1;
      transform: initial;
  }
`;

export const TitleSC = styled.h1`
  color: ${(props) => props.theme.color.primary};
  font-family: ${(props) => props.theme.fontFamily.fontDefault};

  font-size: 4rem;
  margin: 0;
  opacity: 0;
  transform: translateY(-50px);
  animation: ${TextAnimation} 0.5s linear forwards;

  @media (max-width: 870px) {
    font-size: 8vw;
  }
  @media (max-width: 500px) {
    text-align: center;
    font-size: 10vw;
  }
`;
