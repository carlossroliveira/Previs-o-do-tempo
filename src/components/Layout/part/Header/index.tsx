// -------------------------------------------------
// Packages
// -------------------------------------------------
import React from 'react';
// -------------------------------------------------
// Components
// -------------------------------------------------

// -------------------------------------------------
// Styles
// -------------------------------------------------
import { ContainerSC } from './headerStyles';
// -------------------------------------------------
// Types
// -------------------------------------------------

export const Header = (): JSX.Element => {
  return <ContainerSC>HEADER</ContainerSC>;
};
