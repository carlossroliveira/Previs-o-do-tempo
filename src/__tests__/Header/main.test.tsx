// -------------------------------------------------
// Packages
// -------------------------------------------------
import React from 'react';

import 'jest-styled-components';
import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
// -------------------------------------------------
// Components
// -------------------------------------------------
import theme from '../../styles/themes/dark';
import { Header } from '../../components/Layout/part/Header';
// -------------------------------------------------
// Utils
// -------------------------------------------------
import WithThemeComponent from '../utils/withThemeComponent';

const HeaderWithTheme = WithThemeComponent(Header);

describe('Header Testing', () => {
  it('Should render the header component', () => {
    render(<HeaderWithTheme />);
    expect(
      screen.getByRole('heading', {
        name: /previsão do tempo/i,
      }),
    ).toBeInTheDocument();
  });

  it('should render the header component with the right color selected', () => {
    render(<HeaderWithTheme />);
    expect(
      screen.getByRole('heading', {
        name: /previsão do tempo/i,
      }),
    ).toHaveStyle({
      color: theme.color.primary,
    });
  });

  it('should match snapshot', () => {
    const component = render(<HeaderWithTheme />);
    expect(component).toMatchSnapshot();
  });
});
