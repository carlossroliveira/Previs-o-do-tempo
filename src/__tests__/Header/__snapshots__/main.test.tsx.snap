// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Testing should match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c0 {
  grid-area: header;
  margin: 0 0 0 4rem;
}

.c1 {
  color: #EEEEEE;
  font-family: Roboto,sans-serif;
  font-size: 4rem;
  margin: 0;
  opacity: 0;
  -webkit-transform: translateY(-50px);
  -ms-transform: translateY(-50px);
  transform: translateY(-50px);
  -webkit-animation: eQudGy 0.5s linear forwards;
  animation: eQudGy 0.5s linear forwards;
}

@media (max-width:500px) {
  .c0 {
    margin: 0;
  }
}

@media (max-width:870px) {
  .c1 {
    font-size: 8vw;
  }
}

@media (max-width:500px) {
  .c1 {
    text-align: center;
    font-size: 10vw;
  }
}

<body>
    <div>
      <section
        class="c0"
      >
        <h1
          class="c1"
        >
          Previsão do Tempo
        </h1>
      </section>
    </div>
  </body>,
  "container": .c0 {
  grid-area: header;
  margin: 0 0 0 4rem;
}

.c1 {
  color: #EEEEEE;
  font-family: Roboto,sans-serif;
  font-size: 4rem;
  margin: 0;
  opacity: 0;
  -webkit-transform: translateY(-50px);
  -ms-transform: translateY(-50px);
  transform: translateY(-50px);
  -webkit-animation: eQudGy 0.5s linear forwards;
  animation: eQudGy 0.5s linear forwards;
}

@media (max-width:500px) {
  .c0 {
    margin: 0;
  }
}

@media (max-width:870px) {
  .c1 {
    font-size: 8vw;
  }
}

@media (max-width:500px) {
  .c1 {
    text-align: center;
    font-size: 10vw;
  }
}

<div>
    <section
      class="c0"
    >
      <h1
        class="c1"
      >
        Previsão do Tempo
      </h1>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
